### PR TITLE
Add coverage and symbolic link to test regex_check method

### PR DESCRIPTION
**Pull request by szuananwar**
Test_regex_check method add in ../test/builders/test_builders.py file
A symbolic link for ../../tutorials/test_status/status_regex.yml in test/builders

### CODE REVIEW

**Comment by shahzebsiddiqui** [comment]
> Test_regex_check method add in ../test/builders/test_builders.py file A symbolic link for ../../tutorials/test_status/status_regex.yml in test/builders

Please take the time to write a description message instead of the auto-generated description from the commit message. 

Thanks.

**Comment by shahzebsiddiqui** [comment]
@szuananwar can you please remove the `.idea` folder from the git commit. Please see https://github.com/buildtesters/buildtest/pull/1497/files. I think this comes up with PyCharm.  I think a better thing to do is put `.idea` in the top-level .gitignore so it automatically gets ignored. I looked at https://rider-support.jetbrains.com/hc/en-us/articles/207097529-What-is-the-idea-folder- to get an idea of what these files are. They are specific to PyCharm IDE editor for project settings

File: tests/builders/test_builders.py [hunk 1]
```diff
@@ -138,3 +138,12 @@ def test_file_count():
...
     cmd.build()
+    
+def test_regex_check():
+    """This test buildspec using status check with  'status_regex'"""
+    cmd = BuildTest(
+        buildspecs=[os.path.join(here, "status_regex.yml")],
+        buildtest_system=system,
+        configuration=config,
+    )
+    cmd.build()
```
Comment (shahzebsiddiqui) [inline tests/builders/test_builders.py]: this looks good. Take a look at https://app.codecov.io/gh/buildtesters/buildtest/pull/1497/blob/buildtest/buildsystem/checks.py 

In particular the missing coverage for the else statement. Can you please try to add coverage for this. You will want to create a custom buildspec and just simply run it or you can update the main `status_regex.yml`
Screen Shot 2023-06-23 at 9 28 08 AM

**Comment by szuananwar** [comment]
> > Test_regex_check method add in ../test/builders/test_builders.py file A symbolic link for ../../tutorials/test_status/status_regex.yml in test/builders
> 
> Please take the time to write a description message instead of the auto-generated description from the commit message.
> 
> Thanks.



> > Test_regex_check method add in ../test/builders/test_builders.py file A symbolic link for ../../tutorials/test_status/status_regex.yml in test/builders
> 
> Please take the time to write a description message instead of the auto-generated description from the commit message.
> 
> Thanks.

Changed, thanks!

**Comment by shahzebsiddiqui** [comment]
@braeeast2001 please take a look at this PR it will help you address your assigned task

**Comment by shahzebsiddiqui** [comment]
@szuananwar can you please run `buildtest style -a` or see the Style Check error message. 

You will want to make sure you have the precommit setup see https://buildtest.readthedocs.io/en/devel/contributing/code_contribution_guide.html#configuring-black-pre-commit-hook  
Screen Shot 2023-06-23 at 10 59 29 AM

For style check see https://buildtest.readthedocs.io/en/devel/contributing/code_contribution_guide.html#running-stylechecks-via-buildtest-stylecheck just simply run this command and it will work except for `pyflakes` which you need to do manually

**Comment by szuananwar** [comment]
> @szuananwar can you please remove the `.idea` folder from the git commit. Please see https://github.com/buildtesters/buildtest/pull/1497/files. I think this comes up with PyCharm. I think a better thing to do is put `.idea` in the top-level .gitignore so it automatically gets ignored. I looked at https://rider-support.jetbrains.com/hc/en-us/articles/207097529-What-is-the-idea-folder- to get an idea of what these files are. They are specific to PyCharm IDE editor for project settings

Done!

**Comment by shahzebsiddiqui** [comment]
@szuananwar can you please add `.idea` file to the top-level .gitignore so it is excluded

**Review by shahzebsiddiqui — CHANGES_REQUESTED** [review]
Thanks @szuananwar for addressing my previous remarks. I would like you to increase coverage by testing the case where its `stderr` see my message https://github.com/buildtesters/buildtest/pull/1497/files#r1239821776

**Pull request closed without merging**